AmplifyPlugins/API/AWSAPICategoryPlugin/AWSAPIPlugin+Reachability.swift

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright 2018-2019 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Foundation
+import Combine
+import Reachability
+
+extension AWSAPIPlugin {
+
+    @available(iOS 13.0, *)
+    public func reachability(apiName: String, allowsCellularAccess: Bool) -> AnyPublisher<ReachabilityUpdate, Never>? {
+        guard let baseUrl = self.pluginConfig.endpoints[apiName]?.baseURL,
+            let host = baseUrl.host,
+            var reachabilityMap = self.reachabilityMap else {
+                return nil
```

Comment:
some questions and suggestion for debugging:
- apiName: where does it come from? what if apiName cannot be found? how does the caller (assuming datastore) know the apiName? suggestion: separate the logic from retrieving the endpoint by apiName out from this line to consolidate logging for missing endpoint due to bad apiname

- guard var reachabilityMap = self.reachabilityMap else log("missing reachability map"), why is reachabilityMap optional?

Reply:
reachabilityMap is a var ( and not let) because the compiler tells me that computed properties are not not allowed to be 'let' declarations, and it's a computed property because the compiler doesn't let me use @available on a stored property.

Reply:
if you are referring to the local variable reachabilityMap, then, this variable is a var because we are updating a key/value pair down below.